### PR TITLE
Patch attention score far off issue for sd 1.5

### LINKED ISSUE #607: The quality of generated image using NeuronStableDiffusionPipeline is very poor
### System Info

```shell
AMI :Deep Learning AMI Neuron PyTorch 1.13 (Ubuntu 20.04) 20240411
python : 3.8.10

[libraries]
optimum-neuron 0.0.22
torch 2.1.2 
torch-neuronx 2.1.2.2.1.0
torch-xla 2.1.2 
torchvision 0.16.2
transformers 4.36.2
transformers-neuronx 0.10.0.21
triton 2.1.0
diffusers 0.27.2
```

### Who can help?

@JingyaHuang 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

I followed the part "Export via Python API" in [this link](https://huggingface.co/docs/optimum-neuron/tutorials/stable_diffusion#compile-stable-diffusion) to compile the neuron model

```python
from optimum.neuron import NeuronStableDiffusionPipeline
model_id = "runwayml/stable-diffusion-v1-5"
compiler_args = {"auto_cast": "matmul", "auto_cast_type": "bf16"}
input_shapes = {"batch_size": 1, "height": 512, "width": 512}
stable_diffusion = NeuronStableDiffusionPipeline.from_pretrained(model_id, export=True, **compiler_args, **input_shapes)
save_directory = "sd_neuron/"
stable_diffusion.save_pretrained(save_directory)
```

Then I want to compare the generated images between original stable diffusion pipeline and neuron stable diffusion pipeline.
So, I run the code below for the benchmark.
Since cuda is not available in inferentia instance, I run the original stable diffusion pipeline using cpu to set the same generator with same seed.

```python
import torch
import time
def example_prompts():
    prompts = [
        "a photo of an astronaut riding a horse on mars",
        "cute grey cat with blue eyes, wearing a bowtie, acrylic painting",
        "a cute magical flying dog, fantasy art drawn by disney concept artists, highly detailed, digital painting",
        "an illustration of a house with large barn with many cute flower pots and beautiful blue sky scenery",
        "one apple sitting on a table, still life, reflective, full color photograph, centered, close-up product",
        "background texture of stones, masterpiece, artistic, stunning photo, award winner photo",
        "new international organic style house, tropical surroundings, architecture, 8k, hdr",
        "beautiful Renaissance Revival Estate, Hobbit-House, detailed painting, warm colors, 8k, trending on Artstation",
        "blue owl, big green eyes, portrait, intricate metal design, unreal engine, octane render, realistic",
        "delicate elvish moonstone necklace on a velvet background, symmetrical intricate motifs, leaves, flowers, 8k",
    ]

    negative_prompt = "bad composition, ugly, abnormal, malformed"

    return prompts, negative_prompt

prompts, negative_prompt = example_prompts()

def inference(pipe, pipe_type:str):
    NUM_IMAGES_PER_PROMPT = 1
    steps = 30
    seed = 100
    latency_list = []

    for i, prompt in enumerate(prompts):
        inference_start = time.perf_counter()
        g = torch.Generator().manual_seed(seed)
        PIPELINE_GENERATION_CONFIG = {"prompt":prompt,
                                        "negative_prompt":negative_prompt,
                                        "num_inference_steps": steps,
                                        "num_images_per_prompt": NUM_IMAGES_PER_PROMPT,
                                        "guidance_scale": 7.5,
                                        "output_type": "pil",
                                        "generator": g
                                    }
        image = pipe(**PIPELINE_GENERATION_CONFIG).images[0]
        inference_end = time.perf_counter()
        latency = inference_end - inference_start
        latency_list.append(latency)
        print(f"End2End took {latency:.3f} seconds")
        image.save(f"./results/{pipe_type}/{pipe_type}_{i}.png")


inference(stable_diffusion, "optimum")

model_id = "runwayml/stable-diffusion-v1-5"
og_pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float32)
og_pipe = og_pipe.to("cpu")

inference(og_pipe, "og")
```

|   | Prompt | original pipeline | neuron pipeline |
|---|--------|-------------------|-----------------|
| Row 1 | "a cute magical flying dog, fantasy art drawn by disney concept artists, highly detailed, digital painting" | og_2 | optimum_2 |
| Row 2 | "one apple sitting on a table, still life, reflective, full color photograph, centered, close-up product" | og_4 | optimum_4 |
| Row 3 | "blue owl, big green eyes, portrait, intricate metal design, unreal engine, octane render, realistic" | og_8 | optimum_8 |

### Expected behavior

The reason I wrote this post is because I want to use NeuronStableDiffusionPipeline to get the same or very similar results to the images generated by original StableDiffusionPipeline.

Is it common to have a different image than the original StableDiffuisionPipeline when using NeuronStableDiffuisonPipeline?

Or is this a problem that can be solved by adding certain options when compiling the NeuronStableDiffusionPipeline?

Please let me know how to generate the same image when using NeuronStableDiffusionPipeline.

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

Fixes #607

For the SD 1.5 checkpoint, we used to apply optimized attention score (replacing `torch.badbmm` by `torch.bmm`, since we don't have attention_mask as input, it could save some compute). But oddly, the trick works theoretically and works for other SD models, e.g. SD2, and SDXL, but not for SD 1.5...

We will apply in this PR a less optimized `get_attention_scores` for SD1.5 (almost the original one in diffusers) until we find the root cause.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by dacorvo — APPROVED** [review]
LGTM, thanks !

**Review by michaelbenayoun — APPROVED** [review]
